server/handlers/getters.go

```diff
@@ -125,8 +124,7 @@ func (ah *ActivityHandler) getOrCreateSyntheticUser(userID, displayName string)
 		memberUUID := uuid.Parse(userID)
 		encoding := base32.NewEncoding("ybndrfg8ejkmcpqxot1uwisza345h769").WithPadding(base32.NoPadding)
 		shortUserID := encoding.EncodeToString(memberUUID)
-		msDisplayName := slug.Make(userDisplayName)
-		username := "msteams:" + msDisplayName
+		username := "msteams_" + slug.Make(userDisplayName)
```

Comment:
One cool think about the remote users (what we are using for synthetic users) is that we have the possibility of adding `:`, specifically for this cases, so I would use `msteams:` instead of `msteams_`.

Reply:
![image](https://github.com/mattermost/mattermost-plugin-msteams-sync/assets/100013900/9755000b-ac7f-4cb9-8ff3-417b9827c44e)
Adding `:` to the username was not allowed through UI. Also tagging users with `:` in username were not getting tagged. So updated `:` to `_`.
![image](https://github.com/mattermost/mattermost-plugin-msteams-sync/assets/100013900/624ab708-6930-4265-9ec9-8b338ade1a08)

Reply:
It should be a bug, but, yes, lets follow your original approach here, we can decide to change it later. @wiggin77 Do you agree that this is a bug in mattermost side, right? Not being able to mention remote users with `:` in their username, or is on purpose?

Reply:
I'm checking this now, but I suspect it's because the system expects `username:remotename`

Reply:
![remote-user-mention](https://github.com/mattermost/mattermost-plugin-msteams-sync/assets/5024605/4d17edb7-be42-42db-bdae-0098c9864105)

Reply:
@jespino mentions for users with a `username:remotename` are working. See screen above.